src/ApiCompat/Microsoft.DotNet.ApiCompatibility/Rules/AttributesMustMatch.cs

```diff
@@ -0,0 +1,357 @@
+        }
+
+        private void readExclusions(IEnumerable<string> excludeAttributesFiles)
+        {
+            foreach (string filePath in excludeAttributesFiles)
+            {
+                if (!File.Exists(filePath))
+                {
+                    throw new FileNotFoundException($"File {filePath} was not found.", filePath);
+                }
+                foreach (string id in File.ReadAllLines(filePath))
+                {
+                    if (string.IsNullOrWhiteSpace(id) || id.StartsWith("#") || id.StartsWith("//"))
+                    {
+                        continue;
+                    }
+                    _attributesToExclude.Add(id.Trim());
+                }
+            }
+        }
+
+        private void RunOnTypeSymbol(
+            ITypeSymbol? left,
+            ITypeSymbol? right,
+            MetadataInformation leftMetadata,
+            MetadataInformation rightMetadata,
+            IList<CompatDifference> differences)
+        {
+            if (left is null || right is null)
+            {
+                return;
+            }
+
+            // Compare type parameter attributes.
+            if (left is INamedTypeSymbol leftNamed && right is INamedTypeSymbol rightNamed)
+            {
+                if (leftNamed.TypeParameters.Length == rightNamed.TypeParameters.Length)
+                {
+                    for (int i = 0; i < leftNamed.TypeParameters.Length; i++)
+                    {
+                        ReportAttributeDifferences(
+                            left,
+                            left.GetDocumentationCommentId() + $"<{i}>",
+                            leftNamed.TypeParameters[i].GetAttributes(),
+                            rightNamed.TypeParameters[i].GetAttributes(),
+                            differences);
+                    }
+                }
+            }
+
+            ReportAttributeDifferences(
+                left,
+                left.GetDocumentationCommentId() ?? "",
+                left.GetAttributes(),
+                right.GetAttributes(),
+                differences);
+        }
+
+        private void AddDifference(IList<CompatDifference> differences, DifferenceType dt, ISymbol containing, string itemRef, AttributeData attr)
+        {
+            string? docId = attr.AttributeClass?.GetDocumentationCommentId();
+
+            if (docId != null && _attributesToExclude.Contains(docId))
+            {
+                return;
+            }
+
+            CompatDifference difference = dt switch
+            {
+                DifferenceType.Changed => CompatDifference.CreateWithDefaultMetadata(
+                    DiagnosticIds.CannotChangeAttribute,
+                    string.Format(Resources.CannotChangeAttribute, attr.AttributeClass, containing),
+                    DifferenceType.Changed,
+                    itemRef + ":[" + attr.AttributeClass?.GetDocumentationCommentId() + "]"),
```

Comment:
Sorry, I know this was a recent change and I didn't explain it fully. The `CreateWithDefaultMetadata` call is only for tests.

Instead you want to pass the metadata that you receive in the `RunOnMemberSymbol` and `RunOnTypeSymbol` methods down to this methods so that the differences belong to given assemblies.